components/client/src/test/unit/java/com/hotels/styx/client/connectionpool/SimpleConnectionPoolTest.java

```diff
@@ -921,4 +922,98 @@ public void ensureInEstablishmentCountIsDecrementedInError() throws InterruptedE
                 .verifyComplete();
     }
 
+    @Test
+    public void emitsExceptionWhenBrrowingFromClosedPool() {
+
+        SimpleConnectionPool pool = new SimpleConnectionPool(origin, defaultConnectionPoolSettings(), connectionFactory);
+        pool.close();
+        StepVerifier.create(pool.borrowConnection())
+                .expectError(IllegalStateException.class)
+                .verify();
+        assertEquals(pool.stats().connectionAttempts(), 0);
+        assertEquals(pool.stats().busyConnectionCount(), 0);
+        assertEquals(pool.stats().availableConnectionCount(), 0);
+        assertEquals(pool.stats().pendingConnectionCount(), 0);
+    }
+
+    @Test
+    public void justClosesConnectionWhenReturningToClosedPool() {
+        when(connectionFactory.createConnection(any(Origin.class), any(ConnectionSettings.class))).thenReturn(Mono.just(connection1));
+        SimpleConnectionPool pool = new SimpleConnectionPool(origin, defaultConnectionPoolSettings(), connectionFactory);
+
+        StepVerifier.create(pool.borrowConnection())
+                .expectNext(connection1)
+                .verifyComplete();
+        assertEquals(pool.stats().connectionAttempts(), 1);
+        assertEquals(pool.stats().busyConnectionCount(), 1);
+
+        pool.close();
+        assertEquals(0, pool.stats(). closedConnections());
+
+        doAnswer( arg -> {
+            when(connection1.isConnected()).thenReturn(false);
+            return true;
+        }).when(connection1).close();
+        pool.returnConnection(connection1);
+        assertEquals(pool.stats().connectionAttempts(), 1);
+        assertEquals(pool.stats().busyConnectionCount(), 0);
+        assertEquals(1, pool.stats(). closedConnections());
+    }
+
+    @Test
+    public void closesConnectionAndUpdatesMetricsWhenClosingConnectionFromClosedPool() {
+
+        when(connectionFactory.createConnection(any(Origin.class), any(ConnectionSettings.class)))
+                .thenReturn(Mono.just(connection1));
+
+        ConnectionPoolSettings poolSettings = new ConnectionPoolSettings.Builder()
+                .maxConnectionsPerHost(2)
+                .maxPendingConnectionsPerHost(2)
+                .build();
+
+        SimpleConnectionPool pool = new SimpleConnectionPool(origin, poolSettings, connectionFactory);
```

Comment:
Consider inlining the `poolSettings` variable. It is only used once.

Reply:
I think I just copied one of your tests (closesConnection I think) as to keep the code base uniform.  Let me update all the tests that do not require special configuration...